test-utils/src/commonMain/kotlin/mock/resources/MockResources.kt

```diff
@@ -0,0 +1,87 @@
+/*
+ Copyright 2021 Splendo Consulting B.V. The Netherlands
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+ */
+
+package com.splendo.kaluga.test.mock.resources
+
+import com.splendo.kaluga.base.text.format
+import com.splendo.kaluga.resources.Color
+import com.splendo.kaluga.resources.ColorLoader
+import com.splendo.kaluga.resources.Font
+import com.splendo.kaluga.resources.FontLoader
+import com.splendo.kaluga.resources.Image
+import com.splendo.kaluga.resources.ImageLoader
+import com.splendo.kaluga.resources.StringLoader
+
+class MockStringLoader private constructor (
+    private val string: String?,
+    private val quantityString: String?,
+    private val stringMap: Map<String, String?>,
+    private val quantityStringMap: Map<String, String?>
+): StringLoader {
+    constructor(string: String? = null, quantityString: String? = null):
+        this(string, quantityString, emptyMap(), emptyMap())
+    constructor(stringMap: Map<String, String?>, quantityStringMap: Map<String, String?>):
+        this(null, null, stringMap, quantityStringMap)
+
+    override fun loadString(identifier: String, defaultValue: String): String =
+        stringMap[identifier] ?: string ?: defaultValue
+
+    override fun loadQuantityString(
+        identifier: String,
+        quantity: Int,
+        defaultValue: String
+    ): String =
+        (quantityStringMap[identifier] ?: quantityString)?.format(quantity) ?: defaultValue
+}
+
+class MockColorLoader private constructor (
+    private val color: Color?,
+    private val colorMap: Map<String, Color?>
+): ColorLoader {
+    constructor(color: Color? = null): this(color, emptyMap())
+    constructor(colorMap: Map<String, Color?>): this(null, colorMap)
+
+    override fun loadColor(identifier: String, defaultValue: Color?): Color? =
+        colorMap[identifier] ?: color ?: defaultValue
+}
+
+class MockImageLoader private constructor (
+    private val image: Image?,
+    private val imageMap: Map<String, Image?>
+): ImageLoader {
+    constructor(image: Image? = null): this(image, emptyMap())
+    constructor(imageMap: Map<String, Image?>): this(null, imageMap)
```

Comment:
This is of course correct and fine, but I thought you might be interested in a neat Kotlin feature, `Map.withDefault`, this would allow to get rid of the dual field / optional field, as the `constructor(image: Image? = null)` would just pass a  empty map withDefault to the (now) main `constructor(imageMap: Map<String, Image?>)`

Reply:
done

Reply:
was to soon on a conclusion. the complexity grows because `Map.get` wouldn't use the default, and `Map.getValue` would fail when value is not found in a regular map. rolled back